meta-balena-common/recipes-support/os-helpers/os-helpers/os-helpers-tpm2

```diff
@@ -70,27 +70,39 @@ EOF
             if [ -n "${override}" ]; then
                     echo "${override}" | _hexdecode | tcgtool "$var" | _sha256
             else
-                    tcgtool "$var" | _sha256
+                    dd if="${var}" \
+                       status=none \
+                       skip=4 \
+                       bs=1 \
+                    | tcgtool "$var" | _sha256
             fi
         )
 
         digest=$(printf '%s%s' "$digest" "$extend" | _hexdecode | _sha256)
     done
 
     # Extend our digest with any additional random events that may have been logged
+    #
+    # Some systems will log unexpected events before EV_SEPARATOR, some after,
+    # but always before EV_EFI_VARIABLE_AUTHORITY events, so stop there
```

Comment:
Are the events logged after `EV_SEPARATOR` significant? Do they actually affect the hash? I would expect them to be ignored if present, but I can understand this is probably implementation-specific :-/

Reply:
Yes, they do. Every measurement affects the final digest.